### ISSUE TITLE
Placeholder text has default styling

### ISSUE THREAD

**Issue by scurker** [dequelabs/cauldron#1178]
We're not styling placeholder text meaning that it looks odd when set for dark mode.

## Light Mode

![gray placeholder text on white background](https://github.com/dequelabs/cauldron/assets/1062039/5910caa3-7a58-4cba-b10c-ff7a153926a6)

## Dark Mode

![gray placeholder text on dark background](https://github.com/dequelabs/cauldron/assets/1062039/8e7cfa04-6b19-4e94-a41b-8c73d06d8ad0)

We probably should minimally set this for both light/dark modes so that it does look correct is someone uses it.

Change light mode placeholder text to: `#757575` (this should be a new global gray color, `--gray-50`)
Change dark mode placeholder text to: `#8c9fab`

Also add `font-style: italic` to both light/dark modes.

**Comment by yhafez** [comment]
Partially implemented as a new var `--field-placeholder-color` in https://github.com/dequelabs/cauldron/pull/1372. It was applied to placeholders affected by those changes, but we still need to apply it elsewhere
